src/XMakeBuildEngine/BackEnd/BuildManager/BuildParameters.cs

```diff
@@ -320,6 +325,7 @@ private BuildParameters(BuildParameters other)
             _disableInProcNode = other._disableInProcNode;
             _logTaskInputs = other._logTaskInputs;
             _logInitialPropertiesAndItems = other._logInitialPropertiesAndItems;
+            _warningsAsErrors = other._warningsAsErrors == null ? null : new HashSet<string>(other._warningsAsErrors, StringComparer.OrdinalIgnoreCase);
```

Comment:
It's probably not worth changing this, but I would hazard a guess that for this use case (small sets of inputs), a `SortedSet` might actually be faster.